pkg/api/download_test.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2023 The Archivista Contributors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api_test
+
+import (
+	"context"
+	"encoding/json"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"os"
+	"path"
+	"testing"
+
+	"github.com/in-toto/archivista/pkg/api"
+	"github.com/in-toto/go-witness/dsse"
+	"github.com/stretchr/testify/suite"
+)
+
+// Test Suite: UT APIDownload
+type UTAPIDownloadSuite struct {
+	suite.Suite
+}
+
+func TestAPIDownloadSuite(t *testing.T) {
+	suite.Run(t, new(UTAPIDownloadSuite))
+}
+
+func (ut *UTAPIDownloadSuite) Test_Download() {
+
+	testEnvelope, err := os.ReadFile("../../test/package.attestation.json")
+	if err != nil {
+		ut.FailNow(err.Error())
+	}
+	expectedEnvelop := dsse.Envelope{}
+	err = json.Unmarshal(testEnvelope, &expectedEnvelop)
+	if err != nil {
+		ut.FailNow(err.Error())
+	}
+
+	testServer := httptest.NewServer(
+		http.HandlerFunc(
+			func(w http.ResponseWriter, r *http.Request) {
+				w.WriteHeader(http.StatusOK)
+				_, err = w.Write(testEnvelope)
+				if err != nil {
+					ut.FailNow(err.Error())
+				}
+			},
+		),
+	)
+	defer testServer.Close()
```

Comment:
Might be worth a helper function for all these eventually. Especially if the surface area of the API expands more. But, I'm not super worried about it.

Reply:
I'm filing a separate issue for this as an improvement.
We can use some testify fixtures such as setup tests, tear down, etc.

Reply:
- https://github.com/in-toto/archivista/issues/140